### PR TITLE
Use Github Actions for running tests

### PR DESCRIPTION
**Pull request by chaudum**
Documentation is still linted on TravisCI

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,6 +1,2 @@
-pytest-black==0.3.7
```
Comment (christianbader) [inline test-requirements.txt]: Removing this stuff breaks my local setup. Don't really like it.

Comment (chaudum) [reply test-requirements.txt]: In which way? These dependencies have been only installed into tox's virtualenv when running `tox`.

Comment (christianbader) [reply test-requirements.txt]: Yes, sorry for the noise. It never worked for this project 😞

Comment (chaudum) [reply test-requirements.txt]: Ah I think I see your point. I pushed a fixup with `mypy`, `black`, `flake8`, and `isort` dependencies in the `development` extras.

```
$ python -m pip install -Ue ".[develpment"
```

Comment (christianbader) [reply test-requirements.txt]: Thank you!

**Comment by christianbader** [comment]
> What's the reason for not moving the macOS tests and documentation tests to GitHub Actions? OSX is available for GHA: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources

With macOS tests you mean running the tests on a macOS host? Don't think that this is neccessary tbh.

**Comment by chaudum** [comment]
> > What's the reason for not moving the macOS tests and documentation tests to GitHub Actions? OSX is available for GHA: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources
> 
> With macOS tests you mean running the tests on a macOS host? Don't think that this is neccessary tbh.

I would leave that up to the @crate/docs team because they need to maintain it.

**Comment by MarkusH** [comment]
> > > What's the reason for not moving the macOS tests and documentation tests to GitHub Actions? OSX is available for GHA: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources
> > 
> > 
> > With macOS tests you mean running the tests on a macOS host? Don't think that this is neccessary tbh.
> 
> I would leave that up to the @crate/docs team because they need to maintain it.

We're already running tests on macOS, but currently on Travis.

For docs, yeah, leaving that to the docs team seems fine with me.

**Comment by chaudum** [comment]
> We're already running tests on macOS, but currently on Travis.

No, only documentation is linted on macOS
https://travis-ci.org/github/crate/croud/builds/664757215

**Comment by MarkusH** [comment]
> > We're already running tests on macOS, but currently on Travis.
> 
> No, only documentation is linted on macOS
> https://travis-ci.org/github/crate/croud/builds/664757215

:see_no_evil:

**Pull request merged**